benches/bulletproofs.rs

```diff
@@ -112,4 +112,97 @@ criterion_group!{
     targets = verify_rp_8, verify_rp_16, verify_rp_32, verify_rp_64
 }
 
-criterion_main!(create_rp, verify_rp);
+static AGGREGATION_SIZES: [usize; 6] = [1, 2, 4, 8, 16, 32];
+
+fn create_aggregated_rangeproof_helper(n: usize, c: &mut Criterion) {
+    use ristretto_bulletproofs::aggregated_range_proof::SinglePartyAggregator;
+
+    let label = format!("Aggregated {}-bit rangeproof creation", n);
+
+    c.bench_function_over_inputs(
+        &label,
+        move |b, &&m| {
+            let generators = Generators::new(PedersenGenerators::default(), n, m);
+            let mut rng = OsRng::new().unwrap();
+
+            let (min, max) = (0u64, ((1u128 << n) - 1) as u64);
+            let values: Vec<u64> = (0..m).map(|_| rng.gen_range(min, max)).collect();
+
+            b.iter(|| {
+                // Each proof creation requires a clean transcript.
+                let mut transcript = ProofTranscript::new(b"AggregateRangeProofBenchmark");
+
+                SinglePartyAggregator::generate_proof(
+                    &generators,
+                    &mut transcript,
+                    &mut rng,
+                    &values,
+                    n,
+                )
+            })
+        },
+        &AGGREGATION_SIZES,
+    );
+}
+
+fn create_aggregated_rangeproof_n_32(c: &mut Criterion) {
+    create_aggregated_rangeproof_helper(32, c);
+}
+
+fn create_aggregated_rangeproof_n_64(c: &mut Criterion) {
+    create_aggregated_rangeproof_helper(64, c);
+}
+
+fn verify_aggregated_rangeproof_helper(n: usize, c: &mut Criterion) {
+    use ristretto_bulletproofs::aggregated_range_proof::SinglePartyAggregator;
+
+    let label = format!("Aggregated {}-bit rangeproof verification", n);
+
+    c.bench_function_over_inputs(
+        &label,
+        move |b, &&m| {
+            let generators = Generators::new(PedersenGenerators::default(), n, m);
+            let mut rng = OsRng::new().unwrap();
+
+            let (min, max) = (0u64, ((1u128 << n) - 1) as u64);
+            let values: Vec<u64> = (0..m).map(|_| rng.gen_range(min, max)).collect();
+
+            let mut transcript = ProofTranscript::new(b"AggregateRangeProofBenchmark");
+            let proof = SinglePartyAggregator::generate_proof(
+                &generators,
+                &mut transcript,
+                &mut rng,
+                &values,
+                n,
+            ).unwrap();
+
+            b.iter(|| {
+                // Each proof creation requires a clean transcript.
+                let mut transcript = ProofTranscript::new(b"AggregateRangeProofBenchmark");
```

Comment:
I know that the overhead of creating a new transcript is very small, but it's a bit cleaner to clone an existing transcript here, than create one from scratch.